week3/homework/src/index.js

```diff
@@ -1,3 +1,84 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+const Express = require('express');
+const {
+    createTodo,
+    deleteTodo,
+    readTheList,
+    updateTodo,
+    markAsDoneOrNotDone,
+    clearTodos
+} = require('./actions');
+
+const PORT = 3000;
+const app = Express();
+
+// request body parser
+app.use(Express.json());
+
+// createTodo 
+app.post('/todos', (req, res, next) => {
+    const { todo } = req.body;
+    createTodo(todo)
+        .then(data => res.send(data))
+        .catch(err => next(err))
+});
+
+// readList
+app.get('/todos', (req, res) => {
+    readTheList()
+        .then(data => res.send(data))
+        .catch(err => next(err));
+});
+
+// updateTodo
+app.put('/todos/:id', (req, res, next) => {
+    const { id } = req.params;
+    const { todo } = req.body;
+
+    updateTodo(id, todo)
+        .then(data => res.send(data))
+        .catch(err => next(err));
+});
+
+//mark as done
+app.put('/todos/:id/done', (req, res, next) => {
+    const { id } = req.params;
+
+    markAsDoneOrNotDone(id, true)
+        .then(data => res.send(data))
+        .catch(err => next(err));
+});
+
+//mark as undone
+app.put('/todos/:id/undone', (req, res, next) => {
```

Comment:
According to the assignment this should have been `DELETE /todos/:id/done`. But no big deal.